### DISCUSSION TITLE
Pass the terraform plan in github action

### DISCUSSION THREAD

**Discussion by creeper435** [community/community#103088] [Actions] [answered]
### Select Topic Area

Question

### Body

I've built a github action recently for my terraform pipeline which including several main parts:

terraform init
terraform validate
terraform plan

While running the pipeline, I realized that I wasn't commit tfvars file and obviously it failed in terraform plan. So here comes with the question: is there any ways / workaround for me not committing the tfvars file but also make the github action read my tfvars to pass the terraform plan successfully ?

**Comment by gil-son** [comment, marked as answer]
Certainly, you can achieve this by using GitHub Actions secrets to securely store and provide sensitive information, such as your .tfvars file, during the workflow run without committing it to the repository.

Here's a step-by-step to Store tfvars as a GitHub Secret:
1. Go to your GitHub repository.
2. Navigate to the "Settings" tab.
3. Click on "Secrets" in the left sidebar.
4. Click on "New repository secret."
5. Name your secret (e.g., MY_TF_VARS) and paste the content of your .tfvars file into the "Value" field.
6. Click "Add secret" to save.

After that you must to modify your Github Actions Workflow:

```
cli_config: ${{ secrets.MY_TF_VARS }}
```
